home/service/details.py

```diff
@@ -82,13 +84,34 @@ def __init__(self, urn: str):
         self.context = self._get_context()
 
     def _get_context(self):
+        split_datahub_url = urlsplit(
+            os.getenv("CATALOGUE_URL", "https://test-catalogue.gov.uk")
+        )
+
         return {
             "table": self.table_metadata,
             "parent_entity": self.parent_entity,
             "dataset_parent_type": self.dataset_parent_type,
             "h1_value": "Details",
+            "has_lineage": self.get_has_lineage(),
+            "lineage_url": f"{split_datahub_url.scheme}://{split_datahub_url.netloc}/dataset/{self.table_metadata.urn}/Lineage?is_lineage_mode=true&",
         }
 
+    def get_has_lineage(self) -> bool:
+        """
+        Inspects the relationships property of the Table model to establish if a
+        Dataset has any lineage recorded in datahub.
+        """
+        has_lineage = (
```

Comment:
this expression can be shortened to

`len(self.table_metadata.relationships.get(RelationshipType.DATA_LINEAGE, [])) > 0`

(the ternary is just mapping true to true and false to false)